tests/test_core.py

```diff
@@ -109,11 +111,60 @@ def _compare_scaper_jams(jam, regjam):
                 'Unequal values for "{}"'.format(k))
 
     # to compare specs need to covert raw specs to list of lists
-    assert ([[list(x) if isinstance(x, tuple) else x for x in e] for e in
-             ann.sandbox.scaper['bg_spec']] == regann.sandbox.scaper['bg_spec'])
-
-    assert ([[list(x) if isinstance(x, tuple) else x for x in e] for e in
-             ann.sandbox.scaper['fg_spec']] == regann.sandbox.scaper['fg_spec'])
+    # if something in EventSpec is not in the regann, we add it to regann so that 
+    # the test doesn't fail because of an expansion of EventSpec after the
+    # regression data was made. We do this by iterating through the EventSpec tuple
+    # of the generated jams and the regression jams and deletes items in the generated jams 
+    # that don't match with the regression data.
+    
+    bg_spec_list = [[list(x) if isinstance(x, tuple) else x for x in e] for e in
+                    ann.sandbox.scaper['bg_spec']] 
+    fg_spec_list = [[list(x) if isinstance(x, tuple) else x for x in e] for e in
+                    ann.sandbox.scaper['fg_spec']]
+    
+    regann_eventspec_keys = []
+    for e in regann:
+        for k in e.value.keys():
+            regann_eventspec_keys.append(k)
+    regann_eventspec_keys = list(set(regann_eventspec_keys))
+    
+    ann_eventspec_keys = []
+    for e in ann:
+        for k in e.value.keys():
+            ann_eventspec_keys.append(k)
+    ann_eventspec_keys = list(set(ann_eventspec_keys))
+    
+    keys_in_order = ['label', 'source_file', 'source_time', 'event_time', 'event_duration',
+     'snr', 'role', 'pitch_shift', 'time_stretch', 'audio_path']
+    
+    n_fg = 0
+    n_bg = 0
+
+    for a,e in zip(ann,regann):
+        indices = [(keys_in_order.index(k), k)
+                for k in ann_eventspec_keys if k not in regann_eventspec_keys]
+        for index, key in indices:
+            if e.value['role'] == 'foreground':
+                fg_spec_list[n_fg].pop(index)
+            elif e.value['role'] == 'background':
+                bg_spec_list[n_bg].pop(index)
+            a.value.pop(key)
+
+        if e.value['role'] == 'foreground':
+            n_fg += 1
+        if e.value['role'] == 'background':
+            n_bg += 1
```

Comment:
while we dont need it anymore, it is helpful for if changes to the event spec happen at any point in the future. it ensures only testing on keys that are in common between the current event spec and the regression event spec

Reply:
i looked at this again and i dont think the complexity of this change is worth it right now, since EventSpec hasn't changed. so im ditching this. IF we change EventSpec down the road (which is plausible, especially if we're adding more augmentation methods like spatialization or whatever else), then all the old regression data should be tossed. but i think that will go hand-in-hand with the in-memory change.